### PR TITLE
Feature/funding new

### PR DESCRIPTION
**Pull request by GalGend**
A branch for deployment only

### CODE REVIEW

File: client/src/pages/Funding/controller.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	try {
+		const response = await api.getPaymentURL({ ...options });
+		return response;
+	} catch (err) {
+		throw new Error(err);
+	}
+}
+
+export const registerUser = async (options) => {
+	//const requestData = { email };
```
Comment (GalGend) [inline client/src/pages/Funding/controller.js:14]: remove this func

File: client/src/pages/Funding/index.jsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from 'react';
+import { Redirect } from 'react-router-dom'
+import { externalPaymentErrorToast } from 'toasts'
+import YoutubeVideo from 'react-youtube'
+import { Button, Checkbox, TextInput, Divider, HelperText, Link } from '../../shared';
+import { openModal, closeModal } from 'redux/modal/slice'
+import { useDispatch } from 'react-redux'
+import { createPaymentLink, registerUser } from './controller';
```
Comment (GalGend) [inline client/src/pages/Funding/index.jsx:8]: remove the second import 'registerUser'

File: client/src/pages/Funding/index.jsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	const [termsAccepted, setTermsAccepted] = useState(false );
+	const [paymentUrl, setPaymentUrl] = useState();
+	const [onFocusInput, setOnFocusInput] = useState({ name: false, password: false, email: false })
+	const [dirtyInputs, setDirtyInputs] = useState({ name: false, email: false, password: false })
+	const [formErrors, setFormErrors] = useState({
+		amountError:{ isValid: true, message:'' },
+		termsAcceptedError:{ isValid: true, message:'' },
+	})
+
+	const onInputFocus = (inputName) => {
```
Comment (GalGend) [inline client/src/pages/Funding/index.jsx:30]: remove

File: client/src/shared/iframe/iframeUrlChanged.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+function iframeURLChange(iframe, callback) {
+    var lastDispatched = null;
+
+    var dispatchChange = function () {
```
Comment (GalGend) [inline client/src/shared/iframe/iframeUrlChanged.js:4]: remove this file

File: client/src/shared/navigation/desktop.jsx [hunk 5]
```diff
@@ -52,6 +52,11 @@ const DesktopNavBar = ({ user, isAuthenticated, logoutHandler }) => {
 											{t.about}
 										</SC.StyledLink>
 									</Box>
+									<Box px={2}>
+										<SC.StyledLink id="nav-bar-about" to="/funding/" activeClassName="active">
```
Comment (GalGend) [inline client/src/shared/navigation/desktop.jsx:56]: change to the button of "support us" instead of this temp one

**Review by GalGend — COMMENTED** [review]
Ini

File: client/config/default.json [hunk 6]
```diff
@@ -3,8 +3,17 @@
...
         "withCredentials": true
     },
+    "opengraph": {
+        "locale": "he_IL"
+    },
     "geocode": {
         "mapsApiKey": "",
         "autocompleteDelay": 2000
+    },
+    "paymentServices":{
```
Comment (GalGend) [inline client/config/default.json:13]: Remove config no longer needed

File: client/src/pages/Funding/index.jsx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+		try {
+			const paymentpageUrl = await createPaymentLink({amount})
+			setPaymentUrl(paymentpageUrl)
+			dispatch(openModal({ modalType: 'payment', modalProps: { url: paymentpageUrl } }))
+
+		} catch (err) {
+			// error from the paymnet service, or other errors, need to check
+			externalPaymentErrorToast()
+			// if(err.message === 'Error: Request failed with status code 400'){
```
Comment (GalGend) [inline client/src/pages/Funding/index.jsx:57]: Remove commets

File: client/src/pages/Funding/success.jsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+const SuccessPayment = ({ ...props }) => {
+    
+    const notifyClosePage = () => {
+		window.top.postMessage(
+            JSON.stringify({
+              error: false,
+              message: "Close success page"
+            }))
+	}
```
Comment (GalGend) [inline client/src/pages/Funding/success.jsx:15]: הזכות

File: client/config/default.json [hunk 9]
```diff
@@ -3,8 +3,17 @@
         "baseURL": "/api",
         "withCredentials": true
     },
+    "opengraph": {
```
Comment (florpor) [inline client/config/default.json:6]: this also is not needed

File: client/src/locale/he_IL.js [hunk 10]
```diff
@@ -1,6 +1,30 @@
 import React from 'react'
 
 export default {
+	alerts: "התראות",
```
Comment (florpor) [inline client/src/locale/he_IL.js:4]: why are all of these part of this pr?

File: client/src/pages/Funding/validations.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
+export const paymentRequestValidation = ({ amount, termsAccepted }) => {
```
Comment (florpor) [inline client/src/pages/Funding/validations.js:1]: why no semi-colons?

File: client/src/shared/input/index.jsx [hunk 12]
```diff
@@ -8,7 +8,7 @@ import Label from '../label';
 const StyledInput = styled(TextField)`
 	background-color: white;
 	height: 2.75em;
-	width: 24em;
+	width: ${(props) => (props.width || '24em' )} ;
```
Comment (florpor) [inline client/src/shared/input/index.jsx:11]: why the arrow syntax and not just `props.width || '24em'` ?

File: server/api/controller/funding.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
+const Controller = require('./controller');
+const Config = require('../lib/config').paymentServices;
+const axios = require('axios')
+
+let instance = axios.create({
```
Comment (florpor) [inline server/api/controller/funding.js:5]: can you change `axios` to `request-promise` which we already use in the server? (used in iPlanApi.js for example)

File: client/src/pages/Funding/controller.js [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import api from '../../services/paymentApi';
+
+export const createPaymentLink = async (options) => {
+	//const requestData = { email };
+	try {
+		const response = await api.getPaymentURL({ ...options });
+		return response;
+	} catch (err) {
+		throw new Error(err);
```
Comment (florpor) [inline client/src/pages/Funding/controller.js:9]: is this handled properly?
if not, maybe just log and not throw

File: client/src/pages/Funding/controller.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
+import api from '../../services/paymentApi';
+
+export const createPaymentLink = async (options) => {
+	//const requestData = { email };
```
Comment (florpor) [inline client/src/pages/Funding/controller.js:4]: remove comment

**Comment by florpor** [comment]
there are some client build warnings. please take care of those so the client action can succeed

File: server/api/controller/funding.js (hunk 13 above)
Comment (GalGend) [reply server/api/controller/funding.js:5]: @florpor Apparently request-promise is deprecated
image

Comment (GalGend) [reply server/api/controller/funding.js:5]: `bla`

Comment (florpor) [reply server/api/controller/funding.js:5]: ah yeah, i see it was deprecated earlier this year :/ ...
axios it is then

**Pull request merged**